enhancements/sig-policy/99-policy-placement-strategy/README.md

```diff
@@ -0,0 +1,302 @@
+
+- Make `Placement` the primary API for placement (currently the governance propagator is somewhat
+  `PlacementRule`-centric)
+- Leverage the `Placement` helper library to retrieve placement decisions
+- Reflect rollout status per cluster in the root policy for discoverability (whether a cluster is up-to-date or not)
+- Implement the `RolloutStrategy` struct for policies, including:
+  - `RolloutStrategy`
+    - "All": all clusters at once
+    - "Progressive": one cluster at a time
+    - "ProgressivePerGroup": one group of clusters at a time
+  - `Timeout`: Maximum amount of time to wait for success before continuing the rollout
+  - `MandatoryDecisionGroups`: groups that should be handled before other groups
+  - `MaxConcurrency`: Concurrency during a progressive rollout
+- Add an aggregated rollout status for the root policy status.
+
+### Non-Goals
+
+Any specialized behaviors outside of those provided by the `Placement` library and, by extension, the `DecisionStrategy`
+enhancement, that might require additional code other than that already provided/required by the `Placement` library.
+This includes the ability to roll back a rollout. A rollback should be implemented by the user by applying the previous
+version of the policy, and GitOps is the recommended method for this.
+
+## Proposal
+
+### Background
+
+Users create policies on the hub cluster (which are referred to as the root policy). These policies are replicated to
+managed cluster namespaces based on the placement to which they have been bound by the associated `PlacementBinding`
+object. The managed clusters monitor the associated managed cluster namespace on the hub for policy updates.
+
+Additionally, policies have a `remediationAction` specified, which can either be `inform`, meaning the policy controller
+monitors objects on the managed cluster reporting status without making changes, or `enforce`, meaning the policy
+controller attempts to make changes on the managed cluster based on the policy definition and reports status accordingly
+based on the success or failure of the update.
+
+### Design
+
+For this rollout enhancement, policy propagation by the `governance-policy-propagator` controller on the hub would take
+place in a multiple-pass approach:
+
+1. On the first pass following a root policy creation or update, the `governance-policy-propagator` controller on the
+   hub cluster will replicate all policies to managed cluster namespaces and are created regardless of the rollout
+   strategy specified, with the remediation action set to `inform`. This way, all policies are up-to-date and reporting
+   a compliance status based on the current version of the root policy. The aggregated rollout status on the root policy
+   is set to `Progressing`. If the remediation action on the root policy is already `inform`, the rollout status on each
+   cluster is set to `Progressing` and no second pass occurs. If the remediation action is `enforce`, the rollout status
```

Comment:
Does an inform policy that rolls out with some clusters being NonCompliant ever transition out of Progressing to Succeeded?  I think a clarification is needed on how an inform policy rollout can be successful and still identify NonCompliant clusters.

Reply:
It's an interesting question. I was thinking `inform` would follow the `enforce` behaviors, transitioning to `Succeeded` or `Failed` based on the compliance.

But for `inform`, I suppose we could transition the rollout to `Succeeded` once a status is returned regardless of compliance? It'd differ in behavior from `enforce`, but maybe that's okay?

Reply:
Yes, for inform I think as customers learn where problems "may" be they will improve inform policies to detect those issues.  Then the rollout will help identify those problem clusters and I don't think the rollout should fail or be limited because inform policies are working as expected.  Example: Initially we want our addon to just be available.  In v2 we learn we want to look out for terminating config policies....

Reply:
A user shouldn't define a progressive rollout for an inform policy since it will cause the policy to be enforced unless the ConfigurationPolicy has `remediationAction: informOnly` which in that event, the rollout would have no effect.

I think the rollout status should be set to `Succeeded` if any status is provided as Dale mentioned. Syntax errors in the ConfigurationPolicy would lead to a failed rollout status.

Reply:
The nice part of this is that even users that don't use progressive policy rollouts will get a single field to know if all policy templates have reported an up to date status for the cluster.

Reply:
Can I not roll out a policyset that contains inform and enforce policies and the inform policies would not be enforced? I should be able to put any random policies in a policyset with some enforce and some inform and it should be honored appropriately during rollout.  Enforcing some policies could have unintended consequences -- rebooting nodes, upgrading, etc

Reply:
I think it needs more clarification on how the RolloutStatus will move from ToApply->progressing->Succeeded/TimeOut. 
Here is some thoughts;
- **ToApply;** 
     1- when the child policy does not get created yet for the managedCluster
     2- when the child policy is created for the managedCluster **AND** the policy remediation is Inform and the policy in NonCompliant state.
      
- **Progressing;** 
    1- When the child policy is created for the managedCluster **AND** the policy remediation is Enforce and the policy in NonCompliant state.
      
- **Succeeded;** 
   1- When the child policy is created for the managedCluster and it is in Compliant state for either enforce/inform remediation

- **TimeOut;** 
   1- When the child policy is created for the managedCluster  **AND** the policy remediation is Enforce and the policy in NonCompliant state for longer time than it is defined in the rolloutStrategy->TimeOut